### PR TITLE
refactor: remove lien mechanism support

### LINKED ISSUE #8989: Remove lien support from cosmos-sdk
## What is the Problem Being Solved?

Modifications were made to the cosmos-sdk `x/auth` module to support the lien mechanism which has been deprecated and removed from agoric-sdk (swingset side in #7608, golang side in #8988, issue #8987). Since those modifications are no longer needed, we should remove them to reduce our fork footprint and ease future cosmos-sdk upgrades.

## Description of the Design

- Undo the conversion of `AccountKeeper` into an interface, mostly in `x/auth/keeper`
- Remove the `*Action` interfaces for vesting accounts and simplify the implementations of AddGrant, ReturnGrant, Clawback, etc.  Might need to keep `RewardAction`? (`x/auth/vesting/exported`, `x/auth/vesting/types`)
- We might be able to remove the `sdk.Context` argument in `VestingAccount.LockedCoins` back to `time.Time` (`x/auth/vesting`, `x/bank/keeper`).
- Distribution hooks will still be needed for Clawback accounts, but the planned changes for BLD Boost reward interception can be dropped. 

## Security Considerations

Should improve things by getting closer to stock cosmos-sdk.

## Scaling Considerations

N/A

## Test Plan

N/A. Unit tests should suffice.

## Upgrade Considerations

N/A. The account-wrapping feature was designed to not have any impact on stored state.

### PR DESCRIPTION
**Pull request by JimLarson**
# Description

Closes: Agoric/agoric-sdk#8989

- Undo the conversion of `AccountKeeper` into an interface.
- Remove the `*Action` interfaces for vesting accounts and simplify the implementations of AddGrant, ReturnGrant, Clawback, etc.
- Change the sdk.Context argument in VestingAccount.LockedCoins back to time.Time.

Reduces the footprint of our fork.

---

## Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] confirmed `!` in the type prefix if API or client breaking change
* [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] provided a link to the relevant issue or specification
* [ ] reviewed "Files changed" and left comments if necessary
* [x] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#testing)
* [ ] added a changelog entry to `CHANGELOG.md`
* [x] updated the relevant documentation or specification, including comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] confirmed all CI checks have passed

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic, API design and naming, documentation is accurate, tests and test coverage

### CODE REVIEW

**Comment by JimLarson** [comment]
The failing lint check is expected.

File: x/auth/keeper/keeper.go [hunk 1]
```diff
@@ -119,9 +62,9 @@ type accountKeeper struct {
 	addressCdc address.Codec
 }
 
-var _ AccountKeeper = &accountKeeper{}
+var _ AccountKeeperI = &AccountKeeper{}
 
-// NewAccountKeeper returns a new AccountKeeper that uses go-amino to
+// NewAccountKeeper returns a new AccountKeeperI that uses go-amino to
```
Comment (michaelfig) [inline x/auth/keeper/keeper.go:67]:
```suggestion
// NewAccountKeeper returns a new AccountKeeper that uses go-amino to
```

**Review by michaelfig — APPROVED** [review]
LGTM!  I only noticed one search-and-replaceo.

File: x/auth/keeper/keeper.go (hunk 1 above)
Comment (JimLarson) [reply x/auth/keeper/keeper.go:67]: This error is in the cosmos release, so I'm going to err on the side of minimizing our fork footprint. We can fix it in upstream some day.

**Pull request merged**